GitHubReactions.ps1

```diff
@@ -343,15 +394,26 @@ filter Set-GitHubReaction
 
         [Parameter(
             Mandatory,
-            ParameterSetName='ElementsPullRequest',
-            ValueFromPipelineByPropertyName)]
+            ValueFromPipelineByPropertyName,
+            ParameterSetName='ElementsPullRequest')]
         [Parameter(
             Mandatory,
-            ParameterSetName='UriPullRequest',
-            ValueFromPipelineByPropertyName)]
+            ValueFromPipelineByPropertyName,
+            ParameterSetName='UriPullRequest')]
         [Alias('PullRequestNumber')]
         [int64] $PullRequest,
 
+        [Parameter(
+            Mandatory,
+            ValueFromPipelineByPropertyName,
+            ParameterSetName='ElementsComment')]
+        [Parameter(
+            Mandatory,
+            ValueFromPipelineByPropertyName,
+            ParameterSetName='UriComment')]
+        [Alias('CommentId')]
+        [int64] $Comment,
```

Comment:
What's your plan for distinguishing between:

* a CommentId for an Issue Comment
```
POST /repos/:owner/:repo/issues/comments/:comment_id/reactions
```

* a CommentId request for a Pull Request Review Comment
```
GET /repos/:owner/:repo/pulls/comments/:comment_id/reactions
```
?

In both cases, you only need a `CommentId`, and yet they require different endpoints.  I don't see how that gets distinguished with the current function signature.

Reply:
@HowardWolosky it 'just works' because PRs are just Issues. I use the same URL (the issues URL) for both.

Reply:
I added a comment to call this out but from the user's perspective they just need to supply a CommentId which is the only thing `Get-GitHubIssueComment` spits out.

Reply:
Unfortunately, this doesn't appear to be the case.

Let's get all of the PR review comments for this PR:
```powershell
$comments = Invoke-GHRestMethod -UriFragment '/repos/microsoft/PowerShellForGitHub/pulls/193/comments'
```

The very first one in there is [comment ID `433553559`](https://github.com/microsoft/PowerShellForGitHub/pull/193#discussion_r445170775) which I wrote and you reacted to way back when.

If I try to get that reaction from your method, I get a `404`:

```powershell
Get-GitHubReaction -OwnerName microsoft -RepositoryName PowerShellForGitHub -Comment 433553559
# that's calling this endpoint:
# https://api.github.com/repos/microsoft/powershellforgithub/issues/comments/433553559/reactions
```

But if I call the _pull_ comments endpoint, it _does_ work:


```powershell
Invoke-GHRestMethod -UriFragment '/repos/microsoft/PowerShellForGitHub/pulls/comments/433553559/reactions' -Method Get -AcceptHeader 'application/vnd.github.squirrel-girl-preview+json'
# Returns ReactionId 75993900
```

This is something that needs to be sorted out in the function signature now to avoid a breaking change later on.

Reply:
So apparently there's a difference between "Pull Request Comments" and "Pull Request Review Comments". All of your top level PR comments work:

```
Get-GitHubIssueComment -RepositoryName PowerShellForGitHub -OwnerName Microsoft -Issue 193 | Set-GitHubReaction -ReactionType eyes
```

However, Pull Request Review comments do not.

Reply:
Right, that's my concern.  And your function signature at present doesn't have a way to differentiate between the two, so you have no idea which is the right endpoint to call.

Reply:
So here's what I'll do. 

For Get-GitHubIssueComment I've added a PullRequestNumber and IssueNumber  

Get-GitHubReaction will have a parameter set just for Issue (PullRequestNumber will then be alias so that pipeline works) with an optional `Comment` parameter. When we support Pull Request Comments we can split up into 2 parameter sets: 

PullRequestComment 
PullRequestReviewComment 

where we add a ReviewComment param

This would also align with Get-GitHubIssueComment which secretly can get PR Comments today.

Reply:
How about this.  For _this_ PR, let's just remove Comments support.  Everything else about this PR is ready to merge in aside from this issue.  We can _then_ iterate on how to further improve this a natural way with Comments support.

That:
  * Gets this existing work in that you've already spent a lot of time on, unblocking you from doing some of those GitHub Action tests you were interested in
  * Gives us a bit more breathing room to not rush on figuring out the right signature strategy once comments get added.

Reply:
I'm fine waiting. I want to write bots that let you say something like:

@mybot say hello


in a comment, and have it respond.

Reply:
Really -- Smaller changes are always better.  Let's just remove comments from this one and get it in, and then we can have a more focused change with getting Comments working the right way.